lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -218,8 +218,8 @@ class ReportUIFeatures {
   animateHeader() {
     const collapsedHeaderHeight = 50;
     const heightDiff = this.headerHeight - collapsedHeaderHeight + this.headerOverlap;
-    const scrollPct = Math.min(1,
-      this.latestKnownScrollY / (this.headerHeight - collapsedHeaderHeight));
+    const scrollPct = Math.max(0, Math.min(1,
```

Comment:
i always did like devtools' `Number.prototype.constrain` method. :)

Reply:
Funny story, I recently ran into a bizarre case where this pattern was eating up ~300ms of a ~2s execution in another codebase. I say this not to pass any comment on its usage here as it's definitely fine, but as a fun fact in my eternal struggle against @paulirish that perf 🚲 🏠 'ing can be educational and fun :D